src/software/ai/passing/pass_generator.h

```diff
@@ -7,135 +7,65 @@
+        std::shared_ptr<const PassingConfig> passing_config);
 
     /**
-     * Set the id of the robot performing the pass.
+     * Creates a PassEvaluation given a world and a field pitch division.
      *
-     * This id will be used so we ignore the passer when determining where to
-     * pass to. We assume this robot is on the friendly team.
+     * NOTE: It is important that this function is able to run in 10ms.
      *
-     * @param robot_id The id of the robot performing the pass
-     */
-    void setPasserRobotId(unsigned int robot_id);
-
-    /**
-     * Set the target region that we would like to pass to
+     * This is because a pass is evaluated on the provided world and if the
+     * pass generator only completes an iteration of pass updates once every 5 seconds,
+     * very soon when optimizing the pass), and so the passes will likely be invalid by
+     * the time another iteration starts. Because of this, it is extremely important that
+     * the pass generator runs fast enough. Debug builds running on slightly slower
+     * computers could be unable to converge. It is recommended that all testing of things
+     * involving the PassGenerator be done with executables built in "Release" in order to
+     * maximize performance ("Release" can be 2-10x faster then "Debug").
      *
-     * @param area An optional that may contain the area to pass to. If the
-     *             optional is empty (ie. `std::nullopt`) then this indicates
-     *             that there is no target region
-     */
-    void setTargetRegion(std::optional<Rectangle> area);
-
-    /**
-     * Gets the best pass that the PassGenerator has generated so far. The pass generator
-     * may do some work between calls, and getBestPassSoFar will eventually return better
-     * passes over multiple calls.  The best strategy for getting good passes is to call
-     * getBestPassSoFar multiple times, until the pass is rated high enough.
+     * @param world The world to compute the pass evaluation on
      *
      * @return The best currently known pass and the rating of that pass (in [0-1])
      */
-    PassWithRating getBestPassSoFar();
-
-    /**
-     * Destructs this PassGenerator
-     */
-    ~PassGenerator();
+    PassEvaluation<ZoneEnum> generatePassEvaluation(const World& world);
 
 
    private:
     // The number of parameters (representing a pass) that we optimize
-    // (pass_start_x, pass_start_y, pass_speed, pass_start_time)
-    static const int NUM_PARAMS_TO_OPTIMIZE = 4;
-
-    // The number of iterations to run on each call to `getBestPassSoFar()`
-    // **if** running deterministically.
-    // This value was determined to work for unit tests, it will likely need to be
-    // configurable in the future.
-    static const size_t NUM_ITERS_PER_DETERMINISTIC_CALL = 10;
+    // (pass_start_x, pass_start_y, pass_speed)
+    static const int NUM_PARAMS_TO_OPTIMIZE = 3;
```

Comment:
isn't this value predetermined from the definition of Pass in pass.h?

Reply:
yeah it is, its coupled to the `Pass` object

Reply:
shouldn't this constant be defined in the Pass object then?

Reply:
yeah it could be, but it made sense to put it here because its immediately needed by the optimizer

also in `pass.h` it would be a global static const which I was trying to avoid

but I don't feel too strongly about this, so lemme know if you think its better to put it `pass.h`

Reply:
Would this number ever be different from the size of the array in pass.h?

Reply:
Hijacking this to also point out I think the original comment was wrong. This should be the `pass_end_x`, etc. (the end point, not the starting point).

Reply:
No it wouldn't be different 

> Would this number ever be different from the size of the array in pass.h?

Reply:
then I think it should use a constant defined in pass.h

Reply:
moved to pass.h